reccmp/isledecomp/compare/core.py

```diff
@@ -248,68 +247,68 @@ def orig_bin_checker(addr: int) -> bool:
 
             self._db.match_string(string.offset, string.name)
 
-    def _check_if_array_and_match_elements(self, orig_addr: int, name: str):
+    def _match_array_elements(self):
         """
-        Checks if the global variable at `orig_addr` is an array.
+        For each matched variable, check whether it is an array.
         If yes, adds a match for all its elements. If it is an array of structs, all fields in that struct are also matched.
         Note that there is no recursion, so an array of arrays would not be handled entirely.
         This step is necessary e.g. for `0x100f0a20` (LegoRacers.cpp).
         """
 
+        # Helper function
         def _add_match_in_array(
             name: str, type_id: str, orig_addr: int, recomp_addr: int
         ):
+            # pylint: disable=unused-argument
+            # TODO: Previously used scalar_type_pointer(type_id) to set whether this is a pointer
```

Comment:
I can't find an argument why the `compare_type` would be needed. We can always re-introduce it if we need it in the future.